storage/store.go

```diff
@@ -1030,6 +1038,14 @@ func (s *Store) RemoveRange(rng *Range) error {
 	return nil
 }
 
+// PostProcessAfterSnapshot post-processes a range after a Raft
+// snapshot is applied and the range's descriptor has changed. It
+// currently just adds the range to the rangesByKey slice.
+func (s *Store) PostProcessAfterSnapshot(rng *Range) {
+	s.rangesByKey = append(s.rangesByKey, rng)
```

Comment:
If a replica falls too far behind, it may catch up via a snapshot instead of copying the rest of the log, so the same range may receive multiple snapshots. We need to keep track of whether the range has already been added to rangesByKey.

We may want to make this more general than PostProcessAfterSnapshot: something like `ProcessRangeDescriptorUpdate()`, and call it from AdminSplit/AdminMerge/ChangeReplicas as well. None of those currently make any changes that the store needs to know about but it could potentially be useful.

Doesn't this need a lock?